### PR TITLE
GraphGym documentation

### PR DESCRIPTION
**Pull request by JiaxuanYou**
GraphGym documentation, both in Python and Sphinx

### CODE REVIEW

File: torch_geometric/graphgym/utils/tools.py [hunk 1]
```diff
@@ -1,3 +1,14 @@
+import random
+import torch
+import numpy as np
+
+
+def set_seed(seed):
```
Comment (rusty1s) [inline torch_geometric/graphgym/utils/tools.py:6]: There now exists a `seed_everything` function in `torch_geometric` which you can use as well.

Comment (JiaxuanYou) [reply torch_geometric/graphgym/utils/tools.py:6]: Thanks! I'll take a look

Comment (rusty1s) [reply torch_geometric/graphgym/utils/tools.py:6]: See https://github.com/vrtex-team/pytorch_geometric/blob/master/torch_geometric/seed.py

File: docs/source/notes/graphgym.rst [hunk 2]
```diff
@@ -0,0 +1,243 @@
+GraphGym - Manager for experiments
+==================================
+
+**Author: Jiaxuan You**
+
+GraphGym is a platform for **designing and evaluating Graph Neural Networks (GNNs)**.
+GraphGym is originally proposed in `Design Space for Graph Neural Networks, NeurIPS 2020 Spotlight <https://arxiv.org/abs/2011.08843>`__ and hosted in `Github repository <https://github.com/snap-stanford/GraphGym>`__.
+GraphGym has teamed up with PyTorch Geometric to provide an easy-to-use workflow for GNN development.
```
Comment (rusty1s) [inline docs/source/notes/graphgym.rst:8]: Shall we say that GraphGym API is likely going to change as we are continuously working on better and deeper integration with PyG? Similar to how PyTorch is doing it for [`torch.sparse`](https://pytorch.org/docs/stable/sparse.html).

File: docs/source/notes/graphgym.rst [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+
+Highlights
```
Comment (rusty1s) [inline docs/source/notes/graphgym.rst:10]: You have some nice figures in the [original README](https://github.com/snap-stanford/GraphGym). It would be great to add them here as well.

File: docs/source/notes/graphgym.rst [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+**4. Flexible user customization**
+
+- Easily *register your own modules*, such as data loaders, GNN layers, loss functions, etc.
+
+Why GraphGym?
+-------------
+
+**TL;DR:** GraphGym is great for GNN beginners, domain experts and GNN researchers.
+
+**Scenario 1:** You are a beginner to GNN, who wants to understand how GNN works.
```
Comment (rusty1s) [inline docs/source/notes/graphgym.rst:38]:
```suggestion
**Scenario 1:** You are a beginner to GNNs, who wants to understand how GNNs works.
```

File: docs/source/notes/graphgym.rst [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+
+You probably have read many exciting papers on GNN, and try to write your own GNN implementation.
```
Comment (rusty1s) [inline docs/source/notes/graphgym.rst:40]:
```suggestion
You probably have read many exciting papers on GNNs, and try to write your own GNN implementation.
```

File: docs/source/notes/graphgym.rst [hunk 6]
```diff
@@ -0,0 +1,243 @@
...
+Using existing packages for GNN, you still have to code up the essential pipeline on your own.
```
Comment (rusty1s) [inline docs/source/notes/graphgym.rst:41]:
```suggestion
Even if using raw PyTorch Geometric, you still have to code up the essential pipeline on your own.
```
sounds a bit better :)

**Review by rusty1s — APPROVED** [review]

**Comment by rusty1s** [comment]
Before merging, it would be great to fix the CI error caused from building documentation.

File: docs/source/notes/graphgym.rst (hunk 6 above)
Comment (JiaxuanYou) [reply docs/source/notes/graphgym.rst:41]: Indeed! I mostly copy-paste from previous GraphGym docs
